src/dialogs/quick-bar/ha-quick-bar.ts

```diff
@@ -171,11 +225,27 @@ export class QuickBar extends LitElement {
       .sort();
   }
 
-  private async _processItemAndCloseDialog(ev: SingleSelectedEvent) {
+  private async _processCommand(ev: SingleSelectedEvent) {
     const index = ev.detail.index;
     const item = (ev.target as any).items[index].item;
 
-    await this.hass.callService(item.domain, item.service, item.serviceData);
+    this._runCommandAndCloseDialog({
+      domain: item.domain,
+      service: item.service,
+      serviceData: item.serviceData,
+    });
+  }
+
+  private async _runCommandAndCloseDialog(request?: ServiceCallRequest) {
+    if (!request) {
+      return;
+    }
+
+    await this.hass.callService(
+      request.domain,
+      request.service,
+      request.serviceData
+    );
```

Comment:
Since we await this, we should show some progress indicator

Reply:
Here you go! (I put in a `setTimeout(3000)` to simulate a long service call. Actual call is too fast to notice) 

[![Image from Gyazo](https://i.gyazo.com/7e00eb253cfebbfb317d442939397d9d.gif)](https://gyazo.com/7e00eb253cfebbfb317d442939397d9d)

Reply:
Done